plane_wave.ipynb

```diff
@@ -0,0 +1,643 @@
+    "\n",
+    "where $\\mathbf{k}$ indicates the **traveling/propagating direction** of the plane wave when choosing above $\\mathrm{e}^{+\\mathrm{j} \\omega t}$ time convention.\n",
+    "\n",
+    "We figure, that $k_x^2+k_y^2 = \\left(\\frac{\\omega}{c}\\right)^2$ is like Pythagorean theorem for right triangle.\n",
+    "So, we can choose two of the three squared quantities, the third one is then fully determined.\n",
+    "If the (angular) frequency $\\omega$ was already chosen (which we usually start with), we are left to choose only $k_x$ or $k_y$.\n",
+    "This finally leaves us with the choice of radiation angle $\\phi$, which is encoded in the equations\n",
+    "\n",
+    "$$k_x = \\frac{\\omega}{c}\\cos(\\phi)\\qquad k_y = \\frac{\\omega}{c}\\sin(\\phi)$$\n",
+    "\n",
+    "for propagating plane waves (we do not deal here with so called evanescent waves, i.e. we assume $k_x\\in\\mathbb{R}$ and $k_y\\in\\mathbb{R}$).\n",
+    "\n",
+    "Let's check how this can be implemented, plotted and animated to gain some insight into simple wave phenomena."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0e1409a8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "c = 343  # m/s, speed of sound\n",
+    "wave_length = 1  # m\n",
+    "pA = 1  # N / m^2, arbitrarily chosen pressure amplitude\n",
+    "radiation_angle = 0  # deg, propagating direction within xy-plane"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "76d638e9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# square xy-plane\n",
+    "N = 2**6\n",
+    "xylim = 2\n",
+    "x, y = np.meshgrid(np.linspace(-xylim, xylim, N, endpoint=True),\n",
+    "                   np.linspace(-xylim, xylim, N, endpoint=True),\n",
+    "                   indexing='ij')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "103c51d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "w_c = 2*np.pi / wave_length  # rad/m\n",
+    "kx = w_c * np.cos(np.deg2rad(radiation_angle))\n",
+    "ky = w_c * np.sin(np.deg2rad(radiation_angle))\n",
+    "print('dispersion relation ok?', np.allclose(kx**2 + ky**2, w_c**2))\n",
+    "w = w_c*c  # rad/s, angular frequency"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8f39f353",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# plane wave in xy-dimension using exp(+j w t) time convention\n",
+    "p = pA * np.exp(-1j * (kx*x + ky*y))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "035f1dc3",
+   "metadata": {},
+   "source": [
+    "## 1.1 Sound field snapshot using plot_surface"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6c59532e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "col_tick = np.linspace(-pA, +pA, 255, endpoint=True)\n",
```

Comment:
This looks really cool!

My little concern is that the students might understand 2D plane wave as something like a surface wave, which is two different physical phenomena.